mmedit/models/video_interpolater/basic_interpolater.py

```diff
@@ -0,0 +1,223 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import numbers
+import os.path as osp
+
+import mmcv
+from mmcv.runner import auto_fp16
+
+from mmedit.core import psnr, ssim, tensor2img
+from ..base import BaseModel
+from ..builder import build_backbone, build_loss
+from ..registry import MODELS
+
+
+@MODELS.register_module()
+class BasicInterpolater(BaseModel):
+    """Basic model for video interpolation.
+
+    It must contain a generator that takes frames as inputs and outputs a
+    interpolated frame. It also has a pixel-wise loss for training.
+
+    The subclasses should overwrite the function `forward_train`,
+    `forward_test` and `train_step`.
+
+    Args:
+        generator (dict): Config for the generator structure.
+        pixel_loss (dict): Config for pixel-wise loss.
+        train_cfg (dict): Config for training. Default: None.
+        test_cfg (dict): Config for testing. Default: None.
+        pretrained (str): Path for pretrained model. Default: None.
+    """
+    allowed_metrics = {'PSNR': psnr, 'SSIM': ssim}
+
+    def __init__(self,
+                 generator,
+                 pixel_loss,
+                 train_cfg=None,
+                 test_cfg=None,
+                 pretrained=None):
+        super().__init__()
+
+        self.train_cfg = train_cfg
+        self.test_cfg = test_cfg
+
+        # support fp16
+        self.fp16_enabled = False
+
+        # generator
+        self.generator = build_backbone(generator)
+        self.init_weights(pretrained)
+
+        # loss
+        self.pixel_loss = build_loss(pixel_loss)
+
+    def init_weights(self, pretrained=None):
+        """Init weights for models.
+
+        Args:
+            pretrained (str, optional): Path for pretrained weights. If given
+                None, pretrained weights will not be loaded. Defaults to None.
+        """
+        self.generator.init_weights(pretrained)
+
+    @auto_fp16(apply_to=('inputs', ))
+    def forward(self, inputs, target=None, test_mode=False, **kwargs):
```

Comment:
Maybe we can use `input` here. Just like in video SR, we also do not use `lqs` and `gts`. We can simply use singular form here.

Reply:
`input` conflicts with Python's own functions `input()`

Reply:
`inputs` will highlight the difference between input and output

Reply:
Oops yes. Then we can just keep it as it is.